content/en/platform/corda/5.2/deploying-operating/vnodes/bring-your-own-db.md

```diff
@@ -0,0 +1,91 @@
+
+* [api/v5_2/virtualnode/create/db/crypto](../../reference/rest-api/openapi.html#tag/Virtual-Node-API/operation/get_virtualnode_create_db_crypto) — returns the SQL required to create the `crypto` database.
+* [api_v5_2/virtualnode/create/db/uniqueness](../../reference/rest-api/openapi.html#tag/Virtual-Node-API/operation/get_virtualnode_create_db_uniqueness) — returns the SQL required to create the `uniqueness` database.
+* [api/v5_2/virtualnode/create/db/vault/{cpichecksum}](../../reference/rest-api/openapi.html#tag/Virtual-Node-API/operation/get_virtualnode_create_db_vault__cpichecksum_) — returns the SQL required to create the `vault` database for a particular {{< tooltip >}}CPI{{< /tooltip >}} specified by its checksum.
+
+For example, to create the required databases for a virtual node:
+
+1. Retrieve the required SQL for the `crypto` database:
+   {{< tabs >}}
+   {{% tab name="Bash"%}}
+   ```shell
+   curl -k -u $REST_API_USER:$REST_API_PASSWORD -X $REST_API_URL/api/v5_2/virtualnode/create/db/crypto
+   ```
+   {{% /tab %}}
+   {{% tab name="PowerShell" %}}
+   ```shell
+   Invoke-RestMethod -SkipCertificateCheck -Headers @{Authorization=("Basic {0}" -f $REST_API_USER:$REST_API_PASSWORD)} -Method -Uri    $REST_API_URL/virtualnode/create/db/crypto
+   ```
+   {{% /tab %}}
+   {{< /tabs >}}
+2. Retrieve the required SQL for the `uniqueness` database:
+   {{< tabs >}}
+   {{% tab name="Bash"%}}
+   ```shell
+   curl -k -u $REST_API_USER:$REST_API_PASSWORD -X $REST_API_URL/api/v5_2/virtualnode/create/db/uniqueness
+   ```
+   {{% /tab %}}
+   {{% tab name="PowerShell" %}}
+   ```shell
+   Invoke-RestMethod -SkipCertificateCheck -Headers @{Authorization=("Basic {0}" -f $REST_API_USER:$REST_API_PASSWORD)} -Method -Uri    $REST_API_URL/virtualnode/create/db/uniqueness
+   ```
+   {{% /tab %}}
+   {{< /tabs >}}
+3. Retrieve the required SQL for the `vault` database:
+   {{< tabs >}}
+   {{% tab name="Bash"%}}
+   ```shell
+   curl -k -u $REST_API_USER:$REST_API_PASSWORD -X $REST_API_URL/api/v5_2/virtualnode/create/db/vault/<CPI_CHECKSUM>
+   ```
+   {{% /tab %}}
+   {{% tab name="PowerShell" %}}
+   ```shell
+   Invoke-RestMethod -SkipCertificateCheck -Headers @{Authorization=("Basic {0}" -f $REST_API_USER:$REST_API_PASSWORD)} -Method -Uri    $REST_API_URL/virtualnode/create/db/vault/<CPI_CHECKSUM>
+   ```
+   {{% /tab %}}
+   {{< /tabs >}}
+4. Execute the SQL for the three databases.
+
+## Connecting a Virtual Node to Your Own Database
+
+To update a virtual node to connect to your own database use the PUT method of the [/api/v5_2/virtualnode_virtualnodeshortid_db endpoint](../../../reference/rest-api/openapi.html#tag/Virtual-Node-API/operation/put_virtualnode__virtualnodeshortid__db). This method requires the short hash ID of the virtual node as a path parameter and the connection strings as parameters in the request body. For example:
```

Comment:
I think it's more likely this will be used when creating a new virtual node. So maybe we should replace update/PUT with create/POST?

Reply:
Added new steps to virtual node creation pages in Creating Application Networks